src/server/tweetFinder.js

```diff
@@ -1,63 +1,85 @@
- * @deprecated
- * @param response
+ * Searches our database for Tweets and returns them.
+ * If our search is not good enough, access Twitter directly to retrieve more tweets.
+ * @param req A HTTP Request object
+ * @param res A HTTP Response object
+ * @returns {Promise.<T>|*}
  */
-export const exampleDatabaseCall = (request, response) => {
-  db.query('SELECT FROM tweet WHERE content LUCENE :query LIMIT 20',
-    {
-      'params': {
-        'query': request.params.query + '~',
+export const searchQuery = (req, res) => {
+  const query = req.body[0].query;
+  return searchAndCollateResults(query)
+    .then(
+      (data) => {
+        res.writeHead(200, { 'Content-Type': 'application/json' });
+        res.end(JSON.stringify(data));
       },
-    })
-    .then((tweetRecords) => {
-      const result = [];
-
-      const promises = tweetRecords.map((tweetRecord) => {
-        const rid = '' + tweetRecord['@rid'];
-
-        return db.query(
-          `SELECT FROM (TRAVERSE in(TWEETED) FROM (SELECT FROM ${rid})) WHERE @class = "Tweeter"`
-          )
-          .then((tweeterRecords) => {
-            const tweeterRecord = tweeterRecords[0];
-            const toReturn = {
-              'tweet':
-                flattenImmutableObject(
-                  TweetBuilder()
-                    .id(tweetRecord.id)
-                  .content(tweetRecord.content)
-                  .date(tweetRecord.date.toISOString())
-                  .likes(tweetRecord.likes)
-                  .retweets(tweeterRecord.retweets)
-                  .build()),
-              'tweeter':
-                flattenImmutableObject(
-                  TweeterBuilder()
-                    .id(tweeterRecord.id)
-                  .name(tweeterRecord.name)
-                  .handle(tweeterRecord.handle)
-                  .build()),
-            };
-
-            result.push(toReturn);
-          });
-      });
-
-      Promise.all(promises)
-        .then(() => {
-          response.end(
-            JSON.stringify(
-              result));
+
+      (rejection) => {
+        res.writeHead(500, { 'Content-Type': 'application/json' });
+        res.end('An unexpected internal error occurred.');
+        console.warn(`Unable to search for query '${query}'`, rejection);
+      }
+    );
+};
+
+const searchAndCollateResults = (query) => {
+  return searchDatabase(query)
+    .then((data) => {
+      return {
+        'data': {
+          'count': data.length,
+          'records': getTweetsAsResults(data),
+        },
+      };
+    });
+};
+
+const searchDatabase = (query, alreadyAttemptedRefresh = false) => {
+  const tweetSelection = 'SELECT FROM tweet WHERE content LUCENE :query ORDER BY date DESC LIMIT 300';
+
+  return chainPromises(() => {
+    return db.query(tweetSelection, { 'params': { 'query': `${query}~` } });
+  }).then(
+    (tweetRecords) => {
+      const shouldRequeryTwitter = !alreadyAttemptedRefresh && tweetRecords.length <= 20;

```

Comment:
Also are we not doing it by time rather than length of tweets ? ie using moment if most recent is more than 10 mins ago then requery twitter ? 

Or were you going to change it in a later update?